innovation_sweet_spots/pipeline/pilot/investment_predictions/utils.py

```diff
@@ -0,0 +1,323 @@
+import pandas as pd
+import numpy as np
+
+
+def convert_col_to_has_col(df: pd.DataFrame, col: str, drop: bool) -> pd.DataFrame:
+    """Create new column based on another column
+    where non nan values set to 1, nan values to 0
+
+    Args:
+        df: Dataframe to convert columns
+        col: Column to convert
+        drop: Drop original column after converting
+
+    Returns:
+        Dataframe with binarised new column
+    """
+    df[f"has_{col}"] = df[col].notnull().astype("int")
+    if drop:
+        return df.drop(columns=[col])
+    return df
+
+
+def tech_cats_to_dummies(cb_data: pd.DataFrame) -> pd.DataFrame:
+    """Convert tech_category column to dummy columns with values 0 or 1"""
+    combine_tech_cats = (
+        cb_data.groupby("id")
+        .agg({"tech_category": lambda x: ", ".join(map(str, x))})
+        .reset_index()
+    )
+    dummies = combine_tech_cats["tech_category"].str.get_dummies(sep=", ")
+    id_dummies = combine_tech_cats.merge(
+        dummies, left_index=True, right_index=True
+    ).drop(columns=["tech_category"])
+
+    return (
+        cb_data.drop(columns=["tech_category"])
+        .drop_duplicates()
+        .merge(id_dummies, left_on="id", right_on="id")
+    )
+
+
+def dedupe_descriptions(cb_data: pd.DataFrame) -> pd.DataFrame:
+    """For rows which are duplicates except for differences in their description
+    column, drop the rows with no description or the shorter description"""
+    dedupe_descs = (
+        cb_data.groupby("id")
+        .agg(
+            {
+                "description": lambda x: max(
+                    [str(np.nan_to_num(desc, 0)) for desc in x], key=len
+                )
+            }
+        )
+        .reset_index()
+    )
+    return (
+        cb_data.drop(columns=["description"])
+        .drop_duplicates()
+        .merge(dedupe_descs, left_on="id", right_on="id")
+    )
+
+
+def add_unstack_data(
```

Comment:
Very minor comment, but it would be super helpful to provide an example of what this does exactly (ie, input > output). Perhaps either in the docstring (if it's simple to explain) or as a test? 

Don't need to prioritise writing tests at the moment, however, as they take a lot of time - perhaps feel free to add an issue about it.

Reply:
Good idea, I've added an example